### PR TITLE
154 152 implement  _enforce_file_extension() package wide and add defence.py functions to GTFS module

### LINKED ISSUE #152: Implement defence.py functions across the GTFS module
### Description of the Issue  
Currently, some of the GTFS module isn't using functions included in `defence.py`.  
An example of this is in `_order_dataframe_by_day()` (see below)

Image

### LINKED ISSUE #154: Implement _enforce_file_extension() package wide
### Description of the feature to be added  
It is become more common that we are enforcing file extensions throughout various modules. A defence utility has been created for this, therefore it should be used in place of other implementations

### PR DESCRIPTION
**Pull request by cbrowning04**
## Description

This PR:
- Introduces the `defence.py` functions into the GTFS module.
- Implements the `_enforce_file_extensions()` function fro `defence.py` across the package

Fixes #152 
Fixes #154 

## Motivation and Context

To improve the consistency of the defences across the package

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How Has This Been Tested?

Test configuration details:
* OS: Windows 10
* Python version: 3.9.13
* Java version: N/A
* Python management system: conda

## Checklist:

- [x] My code follows the intended structure of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: tests/osm/test_osm_utils.py [hunk 1]
```diff
@@ -70,6 +69,7 @@ def test_filter_osm_defence(self):
         ):
             # type problems with bbox
             filter_osm(bbox=[0, 1.1, 0.1, 1.2])
+        # TODO: Add better testing for OSM on MacOS if possible
```
Comment (r-leyshon) [inline tests/osm/test_osm_utils.py:72]: Good to hear your thoughts on this one @CBROWN-ONS .

Comment (cbrowning04) [reply tests/osm/test_osm_utils.py:72]: Hi @r-leyshon . I can't quite remember too much about this, but I think when I was changing some of the tests, it appeared that some of the features within the `osm` module weren't being tested too thoroughly. Happy to jump on a call to discuss if you'd like.

Comment (r-leyshon) [reply tests/osm/test_osm_utils.py:72]: Comments captured in #191

**Review by r-leyshon — APPROVED** [review]
Great work, good catch with the lint being written to directory on test execution.

**Pull request merged**